### PR TITLE
Feature: Adding, removing and viewing favorite doctors.

### PR DESCRIPTION
**Pull request by kibongos40**
**What does this PR do?**

This PR enhances the existing user interface by integrating backend functionality to manage favorite doctors. Users can now add doctors to their favorites list, remove them, and view their list of favorites directly within the application.

**Description of Task to be completed**

- [x] Favorite a doctor
- [x] View list of favorite doctors
- [x] Remove a doctor from favorites
- [x] Refresh the favorite status icon

How should this be manually tested?

After logging in, to view a list of favorite doctors, you tap on the heart icon beside the notifications icon.
To favorite a doctor, you can do so on the doctor's information screen. You can unfavorite him/her on this screen too.

Screenshots

image
image
image
image

### CODE REVIEW

File: app/Doctors/favoriteDoctors.tsx
Comment (fredshema) [inline app/Doctors/favoriteDoctors.tsx:1]: Consider formatting this file

**Comment by fredshema** [comment]
Also as feedback this can be a text telling a user that they have no favorite doctors instead of an alert

File: app/Doctors/favoriteDoctors.tsx
Comment (kibongos40) [reply app/Doctors/favoriteDoctors.tsx:1]: @fredshema  Done

**Comment by kibongos40** [comment]
> Also as feedback this can be a text telling a user that they have no favorite doctors instead of an alert 


@fredshema Solved
image

**Review by fredshema — APPROVED** [review]

**Review by Emmyfrank — APPROVED** [review]
Well done , I see you worked on all comments

**Pull request merged**